### DISCUSSION TITLE
Ubuntu 22.04 deprecates apt-key; how do I restore access to a repository?

### DISCUSSION THREAD

**Discussion by dsgiigh** [OpenVPN/openvpn3-linux#138] [Q&A - Our support arena] [answered]
Hi, after upgrading to Ubuntu 22.04.1 I got an error when I ran sudo apt update -- a message that apt-key was deprecated and I needed to move to a new method. I removed the openvpn3 repository and tried to reinstall it using these commands:

 sudo wget https://swupdate.openvpn.net/repos/openvpn-repo-pkg-key.pubd

and

sudo apt-key add openvpn-repo-pkg-key.pub

but got this message:

Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).

and there's no sign of the respository in Software & Updates.

I'm unsure how to proceed.

**Comment by flichtenheld** [comment, marked as answer]
https://community.openvpn.net/openvpn/wiki/OpenVPN3Linux already describes the correct method to add repositories without apt-key.

Although despite the warnings apt-key is still fully functional in Ubuntu 22.04 so something else must have gone wrong.

**Reply by dsgiigh** [reply]
Thanks, that solved it. One issue in the instructions to note. The curl commands do not work with sudo. I had to do it in root.
